Docs/source/developers/local_compile.rst

```diff
@@ -0,0 +1,136 @@
+
+Now modify the dependencies as needed in their source locations, update sources if you cloned them earlier, etc.
+When building WarpX, :ref:`the following CMake flags <building-cmake-options>` will use the respective local sources:
+
+.. code-block:: bash
+
+   cd src/warpx
+
+   rm -rf build
+
+   cmake -S . -B build  \
+     -DWarpX_PYTHON=ON  \
+     -DWarpX_amrex_src=$HOME/src/amrex          \
+     -DWarpX_openpmd_src=$HOME/src/openPMD-api  \
+     -DWarpX_picsar_src=$HOME/src/picsar        \
+     -DWarpX_pyamrex_src=$HOME/src/pyamrex      \
+     -DWarpX_pybind11_src=$HOME/src/pybind11
+
+   cmake --build build -j 8
+
+
+.. _developers-local-compile-findpackage:
+
+Compiling With Pre-Compiled Dependencies
+----------------------------------------
+
+This workflow is the best and fastest to compile WarpX, when you just want to change code in WarpX and have the above central dependencies already made available *in the right configurations* (e.g., w/ or w/o MPI or GPU support) from a :ref:`module system <install-hpc>` or :ref:`package manager <install-dependencies>`.
+
+Instead of downloading the source code of the above central dependencies, or using a local copy of their source, we can compile and install those dependencies once.
+By setting the `CMAKE_PREFIX_PATH <https://cmake.org/cmake/help/latest/envvar/CMAKE_PREFIX_PATH.html>`__ environment variable to the respective dependency install location prefixes, we can instruct CMake to `find their install locations and configurations <https://hsf-training.github.io/hsf-training-cmake-webpage/09-findingpackages/index.html>`__.
+
+WarpX supports this with :ref:`the following CMake flags <building-cmake-options>`:
+
+.. code-block:: bash
+
+   cd src/warpx
+
+   rm -rf build
+
+   cmake -S . -B build  \
+     -DWarpX_PYTHON=ON  \
+     -DWarpX_amrex_internal=OFF    \
+     -DWarpX_openpmd_internal=OFF  \
+     -DWarpX_picsar_internal=OFF   \
+     -DWarpX_pyamrex_internal=OFF  \
+     -DWarpX_pybind11_internal=OFF
+
+   cmake --build build -j 8
+
+As a background, this is also the workflow how WarpX is built in :ref:`package managers such as Spack and Conda-Forge <install-dependencies>`.
+
+
+.. _developers-local-compile-pylto:
+
+Faster Python Builds
+--------------------
+
+The Python bindings of WarpX and AMReX (pyAMReX) use `pybind11 <https://pybind11.readthedocs.io>`__.
+Since pybind11 relies heavily on `C++ metaprogramming <https://pybind11.readthedocs.io/en/stable/faq.html#how-can-i-create-smaller-binaries>`__, speeding up the generated binding code requires that we perform a `link-time optimization (LTO) <https://pybind11.readthedocs.io/en/stable/compiling.html#pybind11-add-module>`__ step, also known as `interprocedural optimization (IPO) <https://en.wikipedia.org/wiki/Interprocedural_optimization>`__.
+
+For fast local development cycles, one can skip LTO/IPO with the following flags:
+
+.. code-block:: bash
+
+   cd src/warpx
+
+   cmake -S . -B build       \
```

Comment:
For completeness here, maybe include the setting of the `_src` or `_internal` options here?

Reply:
I would not do that - it is either-or, and people can combine with one of the two options above. Also, the hints here also work perfectly fine and fast with the superbuild, too.